server/src/main/java/org/elasticsearch/action/admin/indices/rollover/RolloverRequest.java

```diff
@@ -67,14 +68,31 @@
             CONDITIONS, ObjectParser.ValueType.OBJECT);
         PARSER.declareField((parser, request, context) -> request.createIndexRequest.settings(parser.map()),
             CreateIndexRequest.SETTINGS, ObjectParser.ValueType.OBJECT);
+        PARSER.declareField((parser, request, includeTypeName) -> {
+            if (includeTypeName) {
+                //expecting one type only
+                for (Map.Entry<String, Object> mappingsEntry : parser.map().entrySet()) {
+                    request.createIndexRequest.mapping((Map<String, Object>) mappingsEntry.getValue());
+                }
+            } else {
+                // a type is not included, add a dummy _doc type
```

Comment:
Rather than doing a check here I think we should just wrap it with `_doc` and pass it on.  If they haven't included `include_type_name` then we're in the typeless world and they shouldn't be including types in their mappings.

Reply:
I basically tried to retrofit the change from 7.x branch.
Could a user set `include_type_name` to false, in this case they are using a compatible only feature - `include_type_name` field but are not using types in a mapping